neurons/validators/penalty/streaming_penalty.py

```diff
@@ -5,11 +5,28 @@
 import bittensor as bt
 from datura.protocol import ScraperStreamingSynapse
 import tiktoken
+import math
 
 
 MAX_TOKENS_PER_CHUNK = 2
 PENALTY_PER_EXCEEDING_TOKEN = 0.01
 
+HARD_TIMEOUT = 10.0  # Hard timeout in seconds
```

Comment:
is wrong, you need to control it by max_execution_time which is used when dendrite is called.

Check models:
Nova
Orbit
Horizon